runbotics/runbotics-orchestrator-ui/src/src-app/components/BotProcessRunner.tsx

```diff
@@ -1,40 +1,65 @@
 import { FC, useEffect, useState } from 'react';
 
 import { LoadingButton } from '@mui/lab';
-import { SvgIcon, Tooltip } from '@mui/material';
+import { IconButton, SvgIcon, Tooltip } from '@mui/material';
 import { unwrapResult } from '@reduxjs/toolkit';
 
 import { useSnackbar } from 'notistack';
-import { Play as PlayIcon, X as XIcon } from 'react-feather';
-import { FeatureKey, IProcess, ProcessInstanceStatus } from 'runbotics-common';
+import {
+    Play as PlayIcon,
+    X as XIcon,
+    RefreshCw as RerunIcon
+} from 'react-feather';
+import {
+    FeatureKey,
+    IProcess,
+    IProcessInstance,
+    ProcessInstanceStatus,
+} from 'runbotics-common';
 import styled from 'styled-components';
 
 import useFeatureKey from '#src-app/hooks/useFeatureKey';
 import useProcessInstanceSocket from '#src-app/hooks/useProcessInstanceSocket';
 import useTranslations from '#src-app/hooks/useTranslations';
 import { useDispatch, useSelector } from '#src-app/store';
-import { processActions, StartProcessResponse } from '#src-app/store/slices/Process';
-import { processInstanceActions, processInstanceSelector } from '#src-app/store/slices/ProcessInstance';
+import {
+    processActions,
+    StartProcessResponse,
+} from '#src-app/store/slices/Process';
+import {
+    processInstanceActions,
+    processInstanceSelector,
+} from '#src-app/store/slices/ProcessInstance';
 
 import { processInstanceEventActions } from '#src-app/store/slices/ProcessInstanceEvent';
-import { schedulerActions, schedulerSelector } from '#src-app/store/slices/Scheduler';
+import { schedulerActions } from '#src-app/store/slices/Scheduler';
 
 import { AttendedProcessModal } from './AttendedProcessModal';
 import If from './utils/If';
 
-
 const BOT_SEARCH_TOAST_KEY = 'bot-search-toast';
 
-const StyledActionButton = styled(LoadingButton)(({ theme }) => `
+const isProcessInstanceFinished = (processInstance: IProcessInstance) => 
+    processInstance?.status === ProcessInstanceStatus.COMPLETED ||
+    processInstance?.status === ProcessInstanceStatus.ERRORED ||
+    processInstance?.status === ProcessInstanceStatus.TERMINATED;
+
+const isProcessActive = (processId: number, processInstance: IProcessInstance) => processId === processInstance?.process.id && !isProcessInstanceFinished(processInstance);
+
+const StyledActionButton = styled(LoadingButton)(
+    ({ theme }) => `
     margin-top: ${theme.spacing(1)};
     & + & {
         margin-left: ${theme.spacing(1)};
     }
-`);
+`
+);
 
 interface BotProcessRunnerProps {
     className?: string;
     process?: IProcess;
+    externalProcessInstance?: IProcessInstance;
```

Comment:
the only purpose of this prop is to perform a rerun so let's give a hint to anyone coming here in the future, I propose `rerunProcessInstance`